webhooks/unifiedstoragequota/validation/unifiedstoragequota_validator.go

```diff
@@ -0,0 +1,332 @@
+	StorageClassName string `json:"storageClassName"`
+
+	// StoragePolicyID represents the StoragePolicyId associated with the given resource
+	StoragePolicyID string `json:"storagePolicyId"`
+
+	// Reason indicates the cause for returning capacity as 0
+	Reason string `json:"reason,omitempty"`
+}
+
+type CapacityResponse struct {
+	RequestedCapacity
+	admission.Response
+}
+
+type RequestedCapacityHandler struct {
+	*pkgctx.WebhookContext
+	admission.Decoder
+
+	Client    client.Client
+	Converter runtime.UnstructuredConverter
+}
+
+// AddToManager adds the webhook to the provided manager.
+func AddToManager(ctx *pkgctx.ControllerManagerContext, mgr ctrlmgr.Manager) error {
+	webhookNameLong := fmt.Sprintf("%s/%s/%s", ctx.Namespace, ctx.Name, webhookName)
+
+	// Build the webhookContext.
+	webhookContext := &pkgctx.WebhookContext{
+		Context:            ctx,
+		Name:               webhookName,
+		Namespace:          ctx.Namespace,
+		ServiceAccountName: ctx.ServiceAccountName,
+		Recorder:           record.New(mgr.GetEventRecorderFor(webhookNameLong)),
+		Logger:             ctx.Logger.WithName(webhookName),
+	}
+	// Initialize the webhook's decoder.
+	decoder := admission.NewDecoder(mgr.GetScheme())
+
+	handler := &RequestedCapacityHandler{
+		Client:         mgr.GetClient(),
+		Converter:      runtime.DefaultUnstructuredConverter,
+		Decoder:        decoder,
+		WebhookContext: webhookContext,
+	}
+	mgr.GetWebhookServer().Register(webhookPath, handler)
+
+	return nil
+}
+
+func (h *RequestedCapacityHandler) Handle(req admission.Request) CapacityResponse {
+	var (
+		obj, oldObj   *unstructured.Unstructured
+		handleRequest func(ctx *pkgctx.WebhookRequestContext) CapacityResponse
+	)
+
+	if req.Operation == v1.Create {
+		obj = &unstructured.Unstructured{}
+		if err := h.DecodeRaw(req.Object, obj); err != nil {
+			return CapacityResponse{Response: webhook.Errored(http.StatusBadRequest, err)}
+		}
+		handleRequest = h.HandleCreate
+	}
+	if req.Operation == v1.Update {
+		obj = &unstructured.Unstructured{}
+		if err := h.DecodeRaw(req.Object, obj); err != nil {
+			return CapacityResponse{Response: webhook.Errored(http.StatusBadRequest, err)}
+		}
+		oldObj = &unstructured.Unstructured{}
+		if err := h.DecodeRaw(req.OldObject, oldObj); err != nil {
+			return CapacityResponse{Response: webhook.Errored(http.StatusBadRequest, err)}
+		}
+		handleRequest = h.HandleUpdate
+	}
+
+	if obj == nil {
+		return CapacityResponse{Response: webhook.Allowed(string(req.Operation))}
+	}
+
+	webhookRequestContext := &pkgctx.WebhookRequestContext{
+		WebhookContext: h.WebhookContext,
+		Op:             req.Operation,
+		Obj:            obj,
+		OldObj:         oldObj,
+		UserInfo:       req.UserInfo,
+		Logger:         h.WebhookContext.Logger.WithName(obj.GetNamespace()).WithName(obj.GetName()),
+	}
+
+	return handleRequest(webhookRequestContext)
+}
+
+func (h *RequestedCapacityHandler) HandleCreate(ctx *pkgctx.WebhookRequestContext) CapacityResponse {
```

Comment:
Can we add a brief comment here and the `HandleUpdate()` function for a better documentation?